backend/Testing/FwHeadless/MediaFileTests.cs

```diff
@@ -0,0 +1,181 @@
+        files.Files.Should().Contain(Path.Join(guid.ToString(), TestRepoZipFilename));
+        await Fixture.PutFile(TestRepoZipPath, guid);
+        files = await Fixture.ListFiles(Fixture.ProjectId);
+        files.Should().NotBeNull();
+        files.Files.Should().Contain(Path.Join(guid.ToString(), TestRepoZipFilename));
+    }
+
+    [Fact]
+    public async Task UploadFile_FailsForNonMembers()
+    {
+        var guid = await Fixture.PostFile(TestRepoZipPath, loginAs: "user", expectSuccess: false);
+        guid.Should().Be(Guid.Empty);
+        var files = await Fixture.ListFiles(Fixture.ProjectId, loginAs: "user");
+        (files?.Files ?? []).Should().NotContain(Path.Join(guid.ToString(), TestRepoZipFilename));
+    }
+
+    [Fact]
+    public async Task UploadFile_WorksForProjectManagers()
+    {
+        var guid = await Fixture.PostFile(TestRepoZipPath, loginAs: "manager", expectSuccess: true);
+        var files = await Fixture.ListFiles(Fixture.ProjectId, loginAs: "manager");
+        (files?.Files ?? []).Should().Contain(Path.Join(guid.ToString(), TestRepoZipFilename));
+    }
+
+    [Fact]
+    public async Task UploadFile_WorksForProjectEditors()
+    {
+        var guid = await Fixture.PostFile(TestRepoZipPath, loginAs: "editor", expectSuccess: true);
+        var files = await Fixture.ListFiles(Fixture.ProjectId, loginAs: "editor");
+        (files?.Files ?? []).Should().Contain(Path.Join(guid.ToString(), TestRepoZipFilename));
+    }
+
+    [Fact]
+    public async Task UploadFile_WithOutMetadata_SizeIsCorrect()
+    {
+        var expectedLength = TestRepoZip.Length;
+        var fileId = await Fixture.PostFile(TestRepoZipPath);
+        var metadata = await Fixture.GetFileMetadata(fileId);
+        metadata.Should().NotBeNull();
+        metadata.SizeInBytes.Should().Be((int)expectedLength);
+    }
+
+    [Fact]
+    public async Task UploadFile_WithMetadata_MetadataIsCorrect()
+    {
+        var expectedLength = TestRepoZip.Length;
+        var uploadMetadata = new FileMetadata
+        {
+            Author = "Test Author",
+            License = MediaFileLicense.CreativeCommons,
+        };
+        var expectedMetadata = new ApiMetadataEndpointResult(uploadMetadata)
+        {
+            Filename = TestRepoZipFilename,
+            SizeInBytes = (int)expectedLength,
+        };
+        expectedMetadata.Author.Should().Be(uploadMetadata.Author);
+        expectedMetadata.License.Should().Be(uploadMetadata.License);
+        var fileId = await Fixture.PostFile(TestRepoZipPath, metadata: uploadMetadata);
+        var metadata = await Fixture.GetFileMetadata(fileId);
+        metadata.Should().NotBeNull();
+        metadata.Should().BeEquivalentTo(expectedMetadata, opts => opts.Excluding(m => m.Sha256Hash));
+    }
+
+    // Filename handling rules:
+    //
+    // 1. If a filename is provided but no file ID (e.g. POST), project ID will be searched for file with same filename (case-SENSITIVE search).
```

Comment:
I actually think we should skip searching for files with the same name, now that there's no name conflict case, we should just create a new file. Otherwise someone might replace a file by accident when they meant to create a new one.

Reply:
I've already removed the code, I just hadn't removed the comment yet.

Done now.